xla/stream_executor/cuda/cuda_blas.cc

```diff
@@ -396,11 +489,19 @@ absl::Status CUDABlas::DoBlasInternalImpl(FuncT cublas_func, Stream *stream,
                                            : CUBLAS_POINTER_MODE_DEVICE)) {
     return absl::InternalError("Failed setting error mode");
   }
+
+  if (capture_logs) CublasLogCollector::On();
```

Comment:
Nit: usually we use RAII to make sure log collection is always turned off on all paths from this function

Reply:
Do you mean a pattern like this?
```c++
if (capture_logs) {
  CublasLogCollector collector;  // constructor enables log capture
  ...
  // destructor disables log capture
} else {
  ...
}
```

Reply:
Implemented `ScopedCublasLogCollection` in a new commit.